### PR TITLE
Readme update

### PR DESCRIPTION
**Pull request by VISHNUVISHAL11**
Readme.md file -fix

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,28 +1,102 @@
...
-## Filter & Search :
 
-Filtering is done by column name while search is performed across columns.
-Multiple values can be supplied at the same time.
+### Problem Statement
 
-`&` -> separates column names
-`|` -> separates multiple values
-`*` -> indicates search across all columns
+Struck with Database with no filtering,searching and pagination features?.
```
Comment (krishnan-techconative) [inline README.md:6]: Datasource not database

File: README.md [hunk 2]
```diff
@@ -1,28 +1,102 @@
...
-1. write logic for filtering
-2. write logic for searching
-3. add enough log statement
-4. extend logic to support nested objects ( filtering, search and sort)
-5. create excception class
-6. Throw exception column name is invalid(if column name is not exist)
-7. Write unit testcases (use cucumber)
-8. document on how to use with example
-9. Finally publish to maven central repo
+### How to use this plugin
```
Comment (krishnan-techconative) [inline README.md:18]: Now you have to include two options on how to use this plugin

**Review by krishnan-techconative — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (VISHNUVISHAL11) [reply README.md:18]: i have changed it .. please look into it

File: README.md (hunk 1 above)
Comment (VISHNUVISHAL11) [reply README.md]: changed ..kindly look into it

File: README.md [hunk 3]
```diff
@@ -1,28 +1,133 @@
-# inmemory-pagination
+# In Memory-POSF
```
Comment (bharathyes) [inline README.md:1]: Would be Inmemory POSF

File: README.md [hunk 4]
```diff
@@ -1,28 +1,133 @@
...
 
-Filtering is done by column name while search is performed across columns.
-Multiple values can be supplied at the same time.
+### Problem Statement
 
-`&` -> separates column names
-`|` -> separates multiple values
-`*` -> indicates search across all columns
+Struck with Datasource with no filtering,searching and pagination features?.
+UI is feeling Hard to handle bulk of data?.
```
Comment (bharathyes) [inline README.md:7]: Stuck not struck
possibly avoid random capitalization. Also `?.`

File: README.md [hunk 5]
```diff
@@ -1,28 +1,133 @@
...
+We provide you the best solution to overcome these issues with minimal overhead.
 
-### Examples:
+### Solution
 
-"name=bharath"
-"*=Vega|vegas"
-"multiMedia.[].name=CCCC&*=Vega|vegas&userId=4051"
+As we are developing so much software in today’s world and dealing with tons of data.
```
Comment (bharathyes) [inline README.md:12]: Sounds like an incomplete sentence when you begin with `as`.

File: README.md [hunk 6]
```diff
@@ -1,28 +1,133 @@
...
+Imagine a situation where you have tons of data to show on your web page.
+A browser will have not able to render all the data at one shot.
```
Comment (bharathyes) [inline README.md:14]: ...will be unable to render...

File: README.md [hunk 7]
```diff
@@ -1,28 +1,133 @@
...
-5. create excception class
-6. Throw exception column name is invalid(if column name is not exist)
-7. Write unit testcases (use cucumber)
-8. document on how to use with example
-9. Finally publish to maven central repo
+### How to use this plugin
+
+##### Approach 1
+
+- To use this plugin , just extend the class **PaginationService** and override the **getRawData()**  function such that it returns list of desired object.
```
Comment (bharathyes) [inline README.md:22]: Most class names have been changed. Please take a look

File: README.md [hunk 8]
```diff
@@ -1,28 +1,133 @@
...
-### Examples:
+### Solution
 
-"name=bharath"
-"*=Vega|vegas"
-"multiMedia.[].name=CCCC&*=Vega|vegas&userId=4051"
+As we are developing so much software in today’s world and dealing with tons of data.
+Imagine a situation where you have tons of data to show on your web page.
+A browser will have not able to render all the data at one shot.
+So the best option is to go for InMemory pagination, where a server will return the bounded number of rows to show in the UI.
```
Comment (bharathyes) [inline README.md:15]: lib name is changed

File: README.md [hunk 9]
```diff
@@ -1,28 +1,133 @@
...
-8. document on how to use with example
-9. Finally publish to maven central repo
+### How to use this plugin
+
+##### Approach 1
+
+- To use this plugin , just extend the class **PaginationService** and override the **getRawData()**  function such that it returns list of desired object.
+- *getRawData()* method will return the complete source data from which the plugin will take care of performing pagination and searching for you.
+
+Sample snippet:
```
Comment (bharathyes) [inline README.md:25]: Instead of repeating `sample snippet` can be tell what the snippet is denotes?

File: README.md [hunk 10]
```diff
@@ -1,28 +1,133 @@
...
+
+https://github.com/techconative/inmemory-posf/blob/536ba7b1062edce986798a96c7c7210302921807/src/test/java/com/techconative/inmemory/pagination/FeedService.java#L23-L34
+
+<br> <br>
+
+##### Approach 2
+
+- Directly call static methods of class InmemoryFOPS Class that the plugin provides by default.
```
Comment (bharathyes) [inline README.md:33]: If possible, reframe this sentence

File: README.md [hunk 11]
```diff
@@ -1,28 +1,133 @@
...
+- it takes List of Map and PaginationCriteria as parameters.
+- Can be used anywhere in your program depending on your use cases.
+
+Sample usage:
+
+```java
+  InmemoryFOPS.applyFiltering( PaginationCriteria criteria, List<Map<String, String>> rawData)
+```
+ 
+- Same way ,we can call *applySorting* and *applyPagination* static method of InmemoryFOPS to leverage its benefits.
```
Comment (bharathyes) [inline README.md:44]: I don't believe so. Those methods are private and should not be accessible from outside the lib

File: README.md [hunk 12]
```diff
@@ -1,28 +1,133 @@
...
+  InmemoryFOPS. applySorting(PaginationCriteria criteria, List<Map<String, String>> rawData);
+  InmemoryFOPS. applyPagination(PaginationCriteria criteria, List<Map<String, String>> rawData);
+```
+<br> <br>
+
+- Plugin provides two class **PaginationCriteria** and **PageResult**  to utilize the  pagination and filtering features on top of the database.
+
+https://github.com/techconative/inmemory-posf/blob/77c1371e8993193dbf0aac299137aec888aa6c56/src/main/java/com/techconative/inmemory/pagination/modal/PaginationCriteria.java#L9-L28
+
+- *PaginationCriteria* helps in setting criteria for pagination depending on your use case.
```
Comment (bharathyes) [inline README.md:57]: Do we need to specifically mention the setter of each member? Can we rather explain what each member variable of that class denote ?

File: README.md (hunk 11 above)
Comment (VISHNUVISHAL11) [reply README.md:44]: i didn't see private .. my bad .. 

have changed it now

File: README.md (hunk 10 above)
Comment (VISHNUVISHAL11) [reply README.md:33]: ok

File: README.md (hunk 8 above)
Comment (VISHNUVISHAL11) [reply README.md:15]: ok

File: README.md (hunk 7 above)
Comment (VISHNUVISHAL11) [reply README.md:22]: ok

File: README.md (hunk 6 above)
Comment (VISHNUVISHAL11) [reply README.md:14]: ok

File: README.md (hunk 5 above)
Comment (VISHNUVISHAL11) [reply README.md:12]: ok

File: README.md (hunk 12 above)
Comment (VISHNUVISHAL11) [reply README.md:57]: i thought method wise will be a better one while implementation

Comment (bharathyes) [reply README.md:48]: getters is a standard format most would be aware of. We just need to explain the purpose of the variable. Eg: `limit` number of results in a page and `pagenumber` selects that page to return after pagination.

Comment (VISHNUVISHAL11) [reply README.md:48]: will change

**Pull request merged**